pennylane_lightning/lightning_qubit/_state_vector.py

```diff
@@ -0,0 +1,386 @@
+            raise ValueError("BasisState parameter and wires must be of equal length.")
+
+        # get computational basis state number
+        basis_states = 2 ** (self.num_wires - 1 - np.array(device_wires))
+        basis_states = qml.math.convert_like(basis_states, state)
+        return int(qml.math.dot(state, basis_states))
+
+    def _apply_state_vector(self, state, device_wires):
+        """Initialize the internal state vector in a specified state.
+        Args:
+            state (array[complex]): normalized input state of length ``2**len(wires)``
+                or broadcasted state of shape ``(batch_size, 2**len(wires))``
+            device_wires (Wires): wires that get initialized in the state
+        """
+
+        if isinstance(state, self._qubit_state.__class__):
+            state_data = allocate_aligned_array(state.size, np.dtype(self.dtype), True)
+            self._qubit_state.getState(state_data)
+            state = state_data
+
+        ravelled_indices, state = self._preprocess_state_vector(state, device_wires)
+
+        # translate to wire labels used by device
+        device_wires = self.map_wires(device_wires)
+        output_shape = [2] * self.num_wires
+
+        if len(device_wires) == self.num_wires and Wires(sorted(device_wires)) == device_wires:
+            # Initialize the entire device state with the input state
+            state = self._reshape(state, output_shape).ravel(order="C")
+            self._qubit_state.UpdateData(state)
+            return
+
+        self._qubit_state.setStateVector(ravelled_indices, state)
+
+    def _apply_basis_state(self, state, wires):
+        """Initialize the state vector in a specified computational basis state.
+
+        Args:
+            state (array[int]): computational basis state of shape ``(wires,)``
+                consisting of 0s and 1s.
+            wires (Wires): wires that the provided computational state should be
+                initialized on
+
+        Note: This function does not support broadcasted inputs yet.
+        """
+        num = self._get_basis_state_index(state, wires)
+        self._create_basis_state(num)
+
+    def _apply_lightning_controlled(self, operation):
+        """Apply an arbitrary controlled operation to the state tensor.
+
+        Args:
+            operation (~pennylane.operation.Operation): operation to apply
+
+        Returns:
+            array[complex]: the output state tensor
+        """
+        state = self.state_vector
+
+        basename = "PauliX" if operation.name == "MultiControlledX" else operation.base.name
+        if basename == "Identity":
+            return
+        method = getattr(state, f"{basename}", None)
+        control_wires = self.wires.indices(operation.control_wires)
+        control_values = (
+            [bool(int(i)) for i in operation.hyperparameters["control_values"]]
+            if operation.name == "MultiControlledX"
+            else operation.control_values
+        )
```

Comment:
```suggestion
        control_values = operation.control_values
```
MultiControlledX now has the control values property.